internal/docker/container.go

```diff
@@ -310,13 +314,15 @@ func (r *ContainerRunner) runJobs(containerOpts <-chan containerStartOptions, re
 	}
 }
 
-func (r *ContainerRunner) collectResults(artifactCfg config.ArtifactDownload, results chan result, expected int) bool {
+func (r *ContainerRunner) collectResults(artifactCfg config.ArtifactDownload, notifierCfg config.Notifications, results chan result, expected int) bool {
 	// TODO find a better way to get the expected
 	completed := 0
 	inProgress := expected
 	passed := true
 
 	junitRequired := report.IsArtifactRequired(r.Reporters, report.JUnitArtifact)
+	slackNtfr, _ := r.Notifier.(*slack.Notifier)
```

Comment:
Code smell 🐛 . What's the point of using the reporter interface if you are doing a forceful cast here?

Reply:
After refactoring, I removed this part.